test/any/Canonicalizer-test.js

```diff
@@ -0,0 +1,181 @@
+      expect(results[0].definition.inheritsFrom).toBeDefined();
+      done();
+    };
+
+    var canonicalizer = Canonicalizer.create({
+      onDone: onDone,
+      waitTime: 3, // Three seconds.
+    });
+
+    canonicalizer.addFragment('Blink', firstAst, 'First file');
+    canonicalizer.addFragment('Blink', secondAst, 'Second file');
+  });
+
+  it('should ignore Enum and Typedef while doing canonicalization', function(done) {
+    // Setting up files for canonicalization.
+    var firstIdl = `
+      typedef (sequence<sequence<ByteString>> or record<ByteString, ByteString>) UnionWithRecord;
+      partial interface SharedWorker {
+        [CallWith=ScriptState, Measure] readonly attribute DOMHighResTimeStamp workerStart;
+      };`;
+    var secondIdl = `
+      enum FoodEnum {
+        "Bread",
+        "Spaghetti",
+        "Sushi"
+      };
+
+      interface SharedWorker : EventTarget {
+        readonly attribute MessagePort port;
+      };
+      interface Test {
+      };`;
+
+    var firstAst = Parser.create().parseString(firstIdl).value;
+    var secondAst = Parser.create().parseString(secondIdl).value;
+
+    // Callback function once Canonicalizer has finished processing.
+    var onDone = function(results) {
+      // Expecting one canonicalized file. (No Enum/Typedef)
+      expect(results.length).toBe(2);
+      // Expecting the file to reference two sources.
+      expect(results[0].ref.length).toBe(2);
+      // Expecting the definition to have 2 members.
+      expect(results[0].definition.members.length).toBe(2);
+      done();
+    };
+
+    var canonicalizer = Canonicalizer.create({
+      onDone: onDone,
+      waitTime: 3, // Three seconds.
+    });
+
+    firstAst.forEach(function(ast) {
+      canonicalizer.addFragment('Blink', ast, 'First file');
+    });
+
+    secondAst.forEach(function(ast) {
+      canonicalizer.addFragment('Blink', ast, 'Second file');
+    });
+  });
+
+  it('should not throw error if two non-partial interfaces were given for different source', function(done) {
+    // Setting up files for canonicalization.
+    var idl = `
+      interface SharedWorker {
+        [CallWith=ScriptState, Measure] readonly attribute DOMHighResTimeStamp workerStart;
+      };`;
+
+    var ast = Parser.create().parseString(idl).value[0];
+    var secondCall = false; // Used for callback tracking.
+
+    // Callback function once Canonicalizer has finished processing.
+    var onDone = function(results) {
+      // Expecting one canonicalized file per source.
+      expect(results.length).toBe(1);
+      if (secondCall) done();
+      else secondCall = true;
+    };
+
+    var canonicalizer = Canonicalizer.create({
+      onDone: onDone,
+      waitTime: 3, // Three seconds.
+    });
+
+    canonicalizer.addFragment('Blink', ast, 'First file');
+    canonicalizer.addFragment('Gecko', ast, 'First file');
+  });
+
+  it('should throw error if two non-partial interfaces were given for same source', function() {
```

Comment:
So far, it seems like most of the cases where two non-partial interfaces are defined, are when they define the exact same members. Currently, repositories have no occurrences of this. It only happens when we canonicalize WebSpec IDL. Thus, I am inclined to believe that it should not normally happen in practice (if we have a good set of data)